torch/distributed/checkpoint/_experimental/checkpointer.py

```diff
@@ -48,21 +123,24 @@ def save(
         state_dict: STATE_DICT,
         path: str,
         **kwargs: Any,
-    ) -> None:
+    ) -> Optional[tuple[Future, Future]]:
         """
-        Save a state dictionary to storage.
-
-        This method writes the state dictionary to storage at the specified path.
-        It ensures that previous checkpoint operations have completed successfully
-        before starting new ones.
+        Save a state dictionary to storage synchronously.
 
         Args:
             state_dict: The state dictionary to save.
             path: The path where the checkpoint should be saved.
             **kwargs: Additional keyword arguments to pass to the writer.
+
+        Returns:
+            Always returns None as operations are synchronous.
+
+        Example:
+            checkpointer.save(state_dict, "/path/to/checkpoint")
         """
-        logger.info("Initiating checkpoint save to %s.", path)
+        logger.debug("Saving checkpoint synchronously to %s", path)
         self._writer.write(state_dict, path, **kwargs)
+        return None
 
```

Comment:
nit add typing.overload to prevent bugs from not overloading the methods during refactor.